runelite-client/src/main/java/net/runelite/client/plugins/minimap/MinimapPlugin.java

```diff
@@ -0,0 +1,155 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.minimap;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import net.runelite.api.Client;
+import net.runelite.api.SpritePixels;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.ResourcesLoaded;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+import javax.inject.Inject;
+import java.awt.Color;
+
+@PluginDescriptor(name = "Minimap")
+public class MinimapPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private MinimapConfig config;
+
+	private Color[] minimapDotColors;
+	private int[][] originalDotColors;
+
+	@Provides
+	private MinimapConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(MinimapConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		originalDotColors = new int[6][20];
+		minimapDotColors = new Color[6];
+
+		resourcesLoaded(null);
+	}
+
+	public void loadColors()
+	{
+		minimapDotColors[0] = config.itemColor();
+		minimapDotColors[1] = config.npcColor();
+		minimapDotColors[2] = config.playerColor();
+		minimapDotColors[3] = config.friendColor();
+		minimapDotColors[4] = config.teamColor();
+		minimapDotColors[5] = config.clanColor();
+	}
+
+	public void storeOriginalDots()
+	{
+		SpritePixels[] originalDots = client.getMapDots();
+
+		if (originalDots == null)
+			return;
+
+		for (int i = 0; i < originalDots.length; ++i)
+		{
+			if (i >= originalDotColors.length)
+				break;
+
+			int[] pixels = originalDots[i].getPixels();
+			for (int j = 0; j < pixels.length; ++j)
+			{
+				originalDotColors[i][j] = pixels[j];
+			}
+		}
+	}
+
+	public void restoreOriginalDots()
+	{
+		SpritePixels[] mapDots = client.getMapDots();
+
+		for (int i = 0; i < mapDots.length; ++i)
+		{
+			if (i >= originalDotColors.length)
+				break;
+
+			for (int j = 0; j < originalDotColors[i].length; ++j)
+			{
+				mapDots[i].getPixels()[j] = originalDotColors[i][j];
+			}
+		}
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		restoreOriginalDots();
+	}
+
+	@Subscribe
+	public void resourcesLoaded(ResourcesLoaded event)
+	{
+		if (client.getMapDots() != null)
+		{
+			storeOriginalDots();
+			loadColors();
+			loadMapDots();
+		}
+	}
+
+	@Subscribe
+	public void configChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals("minimap"))
+			return;
+
+		resourcesLoaded(null);
+	}
+
+	private void loadMapDots()
+	{
+		SpritePixels[] mapDots = client.getMapDots();
+
+		if (mapDots == null)
+			return;
+
+		for (int i = 0; i < mapDots.length; ++i)
+		{
+			if (i >= minimapDotColors.length)
+				break;
+
+			MinimapDot.modify(mapDots[i], minimapDotColors[i]);
```

Comment:
It would be safer to construct your own spritepixels and replace it in the array than to modify the one the client loads